### DISCUSSION TITLE
Listen MessageReactionAdd

### DISCUSSION THREAD

**Discussion by sercanarga** [bwmarrin/discordgo#1168] [Q&A] [answered]
We can add a reaction to the message with ``MessageReactionAdd``. How are we going to listen if the user clicks on the reactions? Does anyone have an example of this?

Thanks.

**Comment by FedorLap2006** [comment, marked as answer]
To listen for a particular event you would need to register an event handler in the `Session` using `Session.AddHandler` method.
`MessageReactionAdd` handler in particular has the following signature: `func (*Session, *MessageReactionAdd)`.
Additionally, since these events received through the [gateway](https://discord.com/developers/docs/topics/gateway), you would need to `Open` a `Session`, if you still haven't done it yet.

**Reply by sercanarga** [reply]
Thank you for your help.
